unirec/main/main.py

```diff
@@ -142,6 +142,7 @@ def get_data_loader(config, task, add_history_trans, MyDataSet, file_path, file_
     dataset = MyDataSet(config, path=file_path, filename=file_name, transform=transform)  
 
     if add_history_trans is not None:
+        add_history_trans = add_history_trans(user2history, config['history_mask_mode'], user2history_time, config['seq_last'], config['data_format'])
```

Comment:
Since "data_type" is added to the init of AddUserHistory, the initialization is moved into get_data_loader, considering that train/valid/test files may have different "data_type".